File: Android_study_demo/CardView/app/src/main/java/com/example/cardview/MainActivity.java
package com.example.cardview;

import android.os.Bundle;
import android.widget.ListView;
import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private ListView mLvMsgList;
    private List<Msg> mDatas = new ArrayList<>();
    private MsgAdapter mAdapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        mLvMsgList = findViewById(R.id.id_lv_msgList);
        mDatas.addAll(MsgLab.generateMockList());
        mAdapter = new MsgAdapter(this,mDatas);
        mLvMsgList.setAdapter(mAdapter);
    }
}

File: Android_study_demo/MallKotlinExample/mall-library/src/main/java/com/mall/library/MainActivity.kt
package com.mall.library

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.mall.library.net.RestClient
import com.mall.library.net.callback.ISuccess

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        RestClient
            .builder()
            .url("")
            .params("", "")
            .success(object :ISuccess{
                override fun onSuccess(response: String) {
                    TODO("Not yet implemented")
                }
            })
            .builder()
            .get()
    }
}